src/Runner.Worker/ExecutionContext.cs

```diff
@@ -364,7 +364,11 @@ public TaskResult Complete(TaskResult? result = null, string currentOperation =
                 }
             }
 
-            _cancellationTokenSource?.Dispose();
+            if (Root != this)
+            {
+                // only dispose TokenSource for step level ExecutionContext 
```

Comment:
when `steprunner` crash with an exception, we complete the job ExecutionContext which causes the token to get disposed, and `Job Finalize` will cause worker crash since it tries to create a new token source from this token.

Reply:
This seems fine. First inclination was a change in JobExtension.FinalizeJob but actually this probably makes more sense (dont dispose since child nodes still should link to get ctrl+c)